source/features/mark-merge-commits-in-list.tsx

```diff
@@ -45,34 +47,36 @@ export function getCommitHash(commit: HTMLElement): string {
 	return getCommitLink(commit)!.pathname.split('/').pop()!;
 }
 
-async function init(): Promise<void> {
-	const isPRConversation = pageDetect.isPRConversation();
-	const pageCommits = $$([
-		'.listviewitem',
-
-		// Old view style (before November 2023)
-		'.js-commits-list-item', // `isCommitList`
-		'.js-timeline-item .TimelineItem:has(.octicon-git-commit)', // `isPRConversation`, "js-timeline-item" to exclude "isCommitList"
-	]);
+const isPRConversation = pageDetect.isPRConversation();
 
-	if (pageCommits.length === 0) {
-		throw new Error('No commits found, selector likely out of date');
+function updateCommitIcon(commit: HTMLElement): void {
+	if (isPRConversation) {
+		// Align icon to the line; rem used to match the native units
+		$('.octicon-git-commit', commit)!.replaceWith(<GitMergeIcon style={{marginLeft: '0.5rem'}}/>);
+	} else {
+		getCommitLink(commit)!.before(<GitMergeIcon className="mr-1"/>);
 	}
+}
 
-	const mergeCommits = await filterMergeCommits(pageCommits.map(commit => getCommitHash(commit)));
-	for (const commit of pageCommits) {
+async function markCommits(commits: HTMLElement[]): Promise<void> {
+	const mergeCommits = await filterMergeCommits(commits.map(commit => getCommitHash(commit)));
+	for (const commit of commits) {
 		if (mergeCommits.includes(getCommitHash(commit))) {
 			commit.classList.add('rgh-merge-commit');
-			if (isPRConversation) {
-				// Align icon to the line; rem used to match the native units
-				$('.octicon-git-commit', commit)!.replaceWith(<GitMergeIcon style={{marginLeft: '0.5rem'}}/>);
-			} else {
-				getCommitLink(commit)!.before(<GitMergeIcon className="mr-1"/>);
-			}
+			updateCommitIcon(commit);
 		}
 	}
 }
 
+async function init(signal: AbortSignal): Promise<void> {
+	observe([
+		'.listviewitem',
+		// Old view style (before November 2023)
+		'.js-commits-list-item', // `isCommitList`
+		'.js-timeline-item .TimelineItem:has(.octicon-git-commit)', // `isPRConversation`, "js-timeline-item" to exclude "isCommitList"
+	], batchedFunction(markCommits, {delay: 100}), {signal});
```

Comment:
Since this works a lot on `getCommitLink`/`getCommitHash`, I think that selector should just be part of this.

```js
const row = [
	'.listviewitem',
	// Old view style (before November 2023)
	'.js-commits-list-item', // `isCommitList`
	'.js-timeline-item .TimelineItem:has(.octicon-git-commit)', // `isPRConversation`, "js-timeline-item" to exclude "isCommitList"
];
const commitLink = [
	'a.markdown-title', // Old view style (before November 2023)
	'.markdown-title a',
];

function init() {
	observe(`:is(${row.join(',')}) ${commitLink.join(',')}`, batchedFunction(markCommits, {delay: 100}), {signal});
}
```

and then

```js
export function getCommitHash(commitTitle: HTMLElement): string {
	return commit!.pathname.split('/').pop()!;
}
async function markCommits(commitTitles: HTMLElement[]): Promise<void> {
 	const mergeCommits = await filterMergeCommits(commitTitles);
 	for (const commit of commitTitles) {
 		if (mergeCommits.includes(getCommitHash(commitTitles))) {
			const commit = commitTitles.closest(commitLink);
 			commit.classList.add('rgh-merge-commit');
			 			if (isPRConversation) {
 				// Align icon to the line; rem used to match the native units
 				$('.octicon-git-commit', commit)!.replaceWith(<GitMergeIcon style={{marginLeft: '0.5rem'}}/>);
 			} else {
 				commitTitle.before(<GitMergeIcon className="mr-1"/>);
 			}
 		}
 	}
 }
```

Reply:
On second thought, no need for this change.